doc/decisions/0010-dynamic-answer-options.md

```diff
@@ -0,0 +1,256 @@
+
+Additional requirements:
+
+- Ability for the label and value to be different
+    - When driven by list collectors, the option's `value` is the `list_item_id` and the `label` is some transformed value on answers stored against the list item id, e.g. `FirstName LastName`.
+        - When driven by answers, the option's `value` should be the answer value, and the `label` should be resolved to the schema label based on the answer id, list item id, and option value.
+        - Supports translated schemas since we are resolving to the label.
+        - Allows for value to be different to the label.
+- Values to resolve correctly when inside and outside of repeats.
+
+## Proposal
+
+Introduce a new `dynamic_options` object that will handle the generation of any dynamic options.
+Existing `options` property to stay as is to provide the ability to define additional options. The additional options will always be appended to the bottom.
+
+Properties and uses:
+
+- `dynamic_options.values`
+    - Used to define the literal values used for [`options.*.value`](https://github.com/ONSdigital/eq-questionnaire-validator/blob/master/schemas/answers/definitions.json#L17-L19).
+    - Structure to allow either a Value Source that returns a sequence or the new Rules structure.
+- `dynamic_options.transform`
+    - Used to apply some transform on each value provided to `dynamic_options.values` which will be used for [`options.*.label`](https://github.com/ONSdigital/eq-questionnaire-validator/blob/master/schemas/answers/definitions.json#L14-L16).
+    - Structure to follow the new Rules structure allowing for the chaining of operators. All transforms are new [value operators](https://github.com/ONSdigital/eq-questionnaire-validator/tree/master/schemas/rules/operators/value).
+    - The reference to `self` within a transform refers to the value of the current iteration.
+      Some mechanism is needed to identify the value of the current iteration and pass it to transforms. For example, in the JSON Logic lib, this is done using `var.""` however, the meaning is not strictly clear, hence `self` was chosen.
+      References to `self` will be resolved before passing the transform in the Rules evaluator. This means the rules can operate as is without any coupling with `dynamic_options`.
+
+### Dynamic answer options with arbitrary sequence (Full week date range)
+
+```json
+{
+  "id": "answer-id",
+  "mandatory": true,
+  "type": "Checkbox",
+  "dynamic_options": {
+    "values": {
+      "map": [
+        {
+          "date-range": [
+            {
+              "date": [
+                {
+                  "source": "response_metadata",
+                  "identifier": "started_at"
+                },
+                {
+                  "days": -7,
+                  "day_of_week": "MONDAY"
+                }
+              ]
+            },
+            7
+          ]
+        },
+        {
+          "format_date": [
```

Comment:
Can the transform not be the argument for the map, rather than the map having a separate operation that simply references the transform? Wouldn't this always resolve to a transform? It stood as as duplication.

Reply:
The transform isn't tied to the map. And it wouldn't always be the same. The `format_date` for map is to return date as `YYYY-MM-DD` for each of the value but the `format_date` in `transform` is for the label which will return as `Monday 11 October 2021` etc. We could allow `transform` to be optional for cases where no transformation is needed.

Reply:
I have updated content to capture that `transforms` can be optional if no transform is needed.

Reply:
Yep, I see that now. Thanks @MebinAbraham